apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoader.java

```diff
@@ -92,38 +90,26 @@
   private static final Logger LOGGER_RECORD_COUNTS =
       LoggerFactory.getLogger(RifLoader.class.getName() + ".recordCounts");
 
-  private final MetricRegistry appMetrics;
   private final LoadAppOptions options;
-  private final HikariDataSource dataSource;
-  private final EntityManagerFactory entityManagerFactory;
   private final IdHasher idHasher;
+  private final PipelineApplicationState appState;
+  private final SecretKeyFactory secretKeyFactory;
 
   /**
    * Constructs a new {@link RifLoader} instance.
    *
-   * @param appMetrics the {@link MetricRegistry} being used for the overall application (as opposed
-   *     to a specific data set)
    * @param options the {@link LoadAppOptions} to use
-   * @param dataSource the {@link DataSource} to use that will be converted to a {@link
-   *     HikariDataSource}
+   * @param appState the {@link PipelineApplicationState} to use
    */
-  public RifLoader(MetricRegistry appMetrics, LoadAppOptions options, DataSource dataSource) {
-    this.appMetrics = appMetrics;
+  public RifLoader(LoadAppOptions options, PipelineApplicationState appState) {
     this.options = options;
-
-    /*
-     * The pool size needs to be double the number of loader threads
-     * when idempotent loads are being used. Apparently, the queries need a
-     * separate Connection?
-     */
-    this.dataSource =
-        DatabaseUtils.createDataSource(dataSource, appMetrics, 2 * options.getLoaderThreads());
-    this.entityManagerFactory = DatabaseUtils.createEntityManagerFactory(dataSource);
+    this.appState = appState;
 
     /*
      * We are re-using the same hash configuration for HICNs and MBIs so we only need one idHasher.
      */
     this.idHasher = new IdHasher(options.getIdHasherConfig());
+    this.secretKeyFactory = createSecretKeyFactory();
```

Comment:
This field is not used anywhere in this class.

Reply:
Good catch -- I goofed this in the merge conflict resolution. Fixed.